Fully-Developed Ice Snowball Summon

Kiran-Koch-Mathews opened the pull request:
I created the Ice Snowball Summon which summons up to four snowball summons that will seek out a nearby hostile target and freeze them for 2 seconds, so they are unable to do damage or move with their AI. They do small jumps similar to a slime and follow the player when idle.

Kiran-Koch-Mathews committed:
non-functional ice summon

Kiran-Koch-Mathews committed:
Ice Summon Update

Updates values, AI Function, and attempts on bug-fixing the projectile.

Kiran-Koch-Mathews committed:
Finished Ice Snowball Summon Art

Updated Ice Snowball Summon Art

Co-Authored-By: Maksim Peterburgskiy <maksim.peterburgskiy@gmail.com>

Kiran-Koch-Mathews committed:
Finished Ice Snowball Summon Weapon

Bug-fixed and cleaned up all the scripts related to the Snowball Summon Weapon, changed the interaction of who the debuff affects and what it does.

Kiran-Koch-Mathews committed:
Main Merge

Kiran-Koch-Mathews committed:
Cleanup

Kiran-Koch-Mathews committed:
Ice Summon 2

Added an upgraded version of the ice summon weapon

Kiran-Koch-Mathews committed:
Summon AI Update and Bug-fixes

I updated the AI so that it will no longer get stuck, I also bug-fixed areas where the PermaFreezeBuff or upgraded Snowball Summon would display incorrectly.

MaksimPeterburgskiy committed:
initial BlizzardBlower commit

MaksimPeterburgskiy committed:
Added invisible projectile for BlizzardBlower
Added functionality code for Blizzard Blower

K3LV0N reviewed (changes requested):
Pull or something first. You have conflicting .vs files which shouldn't be here

MaksimPeterburgskiy committed:
Added better bullet Ai and smoothed out particle stream

Kiran-Koch-Mathews committed:
Updated Advanced Summon Behavior and Bug-fixes

I updated the Advanced Summon's behavior to now have a minimum freeze time and then after that freeze time be disabled once the NPC is too far away or gets hit by the player.

I also updated the sprites of the advanced summon, its recipe list, and bug-fixes some dead pixels.

I also adjusted the names of the items in the localization.

Co-Authored-By: Maksim Peterburgskiy <maksim.peterburgskiy@gmail.com>

Kiran-Koch-Mathews commented:
Added Final small adjustments before the pull;

MaksimPeterburgskiy committed:
Add Inferno Blower

K3LV0N committed:
Added Wheely, the first naturally spawning enemy to the mod.

Also fixed some format issues, added increased generalization for future expansion, and documentation.

K3LV0N committed:
Merge pull request #20 from K3LV0N/main

Merge pull request #17 from K3LV0N/kyle

MaksimPeterburgskiy committed:
Comments

K3LV0N committed:
Merge pull request #21 from K3LV0N/kyle

Kyle

MaksimPeterburgskiy committed:
Update weapon properties and tooltips

MaksimPeterburgskiy committed:
Moved Files

Kiran-Koch-Mathews committed:
Jetpack Ice Summon

I added another upgrade to my ice summons. This one has a jetpack, so it will fly towards the enemies instead of jump as this was part of the weakness of the previous summons, hence an upgrade. I also did some other small balance changes.

Co-Authored-By: Maksim Peterburgskiy <maksim.peterburgskiy@gmail.com>

K3LV0N committed:
Added upgraded versions of Ice bow (now renamed to Glacial Bow)

-All "Ice" items and projectiles are now replaced with "Glacial" to fit the mod more and to make sure we have no vanilla conflicts

-Added additional code formatting (better file structure, and upkeep methods)

K3LV0N committed:
Merge pull request #24 from K3LV0N/kyle

Added upgraded versions of Ice bow (now renamed to Glacial Bow)

K3LV0N committed:
Merge branch 'main' into maksim

K3LV0N committed:
Merge pull request #22 from K3LV0N/maksim

Inferno Blower and Blizzard Blower

K3LV0N committed:
Fixed a bug with Wheely's AI

K3LV0N committed:
Merge pull request #25 from K3LV0N/bowUpgrades2

Fixed a bug with Wheely's AI

Kiran-Koch-Mathews committed:
Limited Flying Time

I added it so that the jetpack does not work unconditionally. It will only be on for 10 seconds, and then it falls to the ground and has to re-charge for 2 seconds first. I then gave it a bit of a buff also where it applies Frostburn

K3LV0N committed:
Fixed some bugs concerning namespace

K3LV0N committed:
Merge pull request #26 from K3LV0N/bowUpgrades2

Fixed some bugs concerning namespace

Kiran-Koch-Mathews committed:
Start of Revive

Reviving my summons once they die after a certain amount of time

Kiran-Koch-Mathews committed:
Start

Kiran-Koch-Mathews committed:
Fixed naming error

Kiran-Koch-Mathews committed:
Added Respawning and Animations

I cleaned up how the buffs looked as I also added a section where it now respawns after a certain amount of seconds of being dead so you do not have to pull out the staff and respawn it.

I also with the help of Maksim, added animations to the jetpack slime so it loses its flame whenever it does not fly

Co-Authored-By: Maksim Peterburgskiy <maksim.peterburgskiy@gmail.com>
